website/docs/cloud-docs/workspaces/settings/vcs.mdx

```diff
@@ -4,81 +4,94 @@ page_title: VCS Connections - Workspaces - Terraform Cloud and Terraform Enterpr
+-> **Note:** Terraform Cloud triggers runs on all attached workspaces if it does not receive a list of changed files or if that list is too large to process. When this happens, Terraform Cloud may show several runs with completed plans that do not result in infrastructure changes.
 
-### Error Handling for Run Triggers
+### VCS Branch
 
-Terraform Cloud retrieves the changed files for each push or pull request using your VCS provider's API. If for some reason the list of changed files cannot be retrieved, or if it is too large to process, the default behavior is to trigger runs on all attached workspaces. Should this happen, you may see several runs with state "Planned", due to the push resulting in no changes to infrastructure.
+This designates which branch of the repository Terraform Cloud should use in the workspace. If left blank, Terraform Cloud uses the repository's default branch.
 
-## Automatic Speculative Plans
+### Automatic Speculative Plans
 
-Whether to perform [speculative plans on pull requests](/cloud-docs/run/ui#speculative-plans-on-pull-requests) to the connected repository, to assist in reviewing proposed changes.
+Whether to perform [speculative plans on pull requests](/cloud-docs/run/ui#speculative-plans-on-pull-requests) to the connected repository, to assist in reviewing proposed changes. Automatic speculative plans are enabled by default, but you can disable them for any workspace.
 
-Automatic speculative plans are enabled by default, but you can disable them for any workspace.
+### Include Submodules on Clone
 
-## VCS Branch
+Select **Include submodules on clone** to recursively clone all of the repository's Git submodules when Terraform Cloud feteches a configuration.
 
-Which branch of the repository to use. If left blank, Terraform Cloud will use the repository's default branch.
+-> **Note:** The [SSH key for cloning Git submodules](/cloud-docs/vcs/#ssh-keys) is set in the VCS provider settings for the organization, and is not related to the workspace's SSH key for Terraform modules.
 
-## Include submodules on clone
+## Glob Patterns for Automatic Run Triggering
 
-Whether to recursively clone all of the repository's Git submodules when fetching a configuration.
+We support `glob` patterns to describe a set of triggers if `trigger patterns` option is selected.
 
--> **Note:** The [SSH key for cloning Git submodules](/cloud-docs/vcs/#ssh-keys) is set in the VCS provider settings for the organization, and is not related to the workspace's SSH key for Terraform modules.
+Supported wildcards:
+`*`  Matches zero or more characters.
+`?`  Matches one or more characters.
+`**` Matches directories recursively.
+
+Examples:
+/**/* matches every file in every directory
```

Comment:
can these be code formatting? and should they be bullets?

Reply:
5ea81af788e02f6e212d1c75ee8d2f0d504e4271